### PR TITLE
✅ (testing): added tests for Request Access and Access Status

### PR DESCRIPTION
**Pull request by FaraiQC**

### CODE REVIEW

File: libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts [hunk 1]
```diff
@@ -15,4 +15,23 @@ describe('AccessStatusService', () => {
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  it('should be not be null',async () => {
+    const call = await service.getAccessStatus("u12345678","42");
+    console.log(call);
```
Comment (arulagrawal) [inline libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts:21]: Should we be cluttering the logs with objects from our tests? @arneschreuder

**Review by arulagrawal — CHANGES_REQUESTED** [review]
Please remove unused import from [libs/api/request-access/service/feature/src/lib/request-access.service.ts](https://github.com/COS-301/graduates/pull/740/files#diff-58cad57ef32cc5c099cff90bd2169d5c552c8a22cd09439105ec99926d6894d0).

Also I think we should be running code formatters or something before we commit and make PRs.

File: libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts (hunk 1 above)
Comment (arulagrawal) [reply libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts:21]: Also as a testing methodology question, is this a good test? The function getAccessStatus() can never return null, so is this a sanity check type thing?

Comment (FaraiQC) [reply libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts:21]: That was for me to use when debugging, I forgot to remove them

Comment (FaraiQC) [reply libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts:21]: let me do that real quick

Comment (arulagrawal) [reply libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts]: > Also as a testing methodology question, is this a good test? The function getAccessStatus() can never return null, so is this a sanity check type thing?

I guess this is fine since it covers the possibility that the function may change in the future?

**Comment by FaraiQC** [comment]
> Please remove unused import from [libs/api/request-access/service/feature/src/lib/request-access.service.ts](https://github.com/COS-301/graduates/pull/740/files#diff-58cad57ef32cc5c099cff90bd2169d5c552c8a22cd09439105ec99926d6894d0).
> 
> Also I think we should be running code formatters or something before we commit and make PRs.

I am using all the imports. I added the CQRS import today because of the pull request that I made in the morning that was not passing a test

By formatters, do you mean like the Prettier extension on vs code? because I am using that

File: libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts (hunk 1 above)
Comment (FaraiQC) [reply libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts]: Yeah just to make sure all bases are covered

File: libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts [hunk 2]
```diff
@@ -15,4 +15,20 @@ describe('AccessStatusService', () => {
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  it('should be not be null',async () => {
```
Comment (arulagrawal) [inline libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts:19]:
```suggestion
  it('should be not be null', async () => {
```
All of these should have spaces like this.

**Review by arulagrawal — CHANGES_REQUESTED** [review]
Not sure how Prettier/VS code isn't picking this up.

File: libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts [hunk 3]
```diff
@@ -15,4 +15,20 @@ describe('AccessStatusService', () => {
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  it('should be not be null', async () => {
+    const call = await service.getAccessStatus("u12345678","42");
+    expect(call).not.toBeNull();
+  });
+
+  it('should be equal to entity',async () => {
```
Comment (arulagrawal) [inline libs/api/access-status/service/feature/src/lib/access-status.service.spec.ts:24]: What

File: libs/api/request-access/service/feature/src/lib/request-access.service.spec.ts [hunk 4]
```diff
@@ -17,4 +17,15 @@ describe('RequestAccessService', () => {
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  it('should be not be null',async () => {
```
Comment (arulagrawal) [inline libs/api/request-access/service/feature/src/lib/request-access.service.spec.ts:21]: these

File: libs/api/request-access/service/feature/src/lib/request-access.service.spec.ts [hunk 5]
```diff
@@ -17,4 +17,15 @@ describe('RequestAccessService', () => {
...
+    const call = await service.getAccessEntity("u12345678","42","5");
+    expect(call).not.toBeNull();
+  });
+
+  it('should be equal to entity',async () => {
```
Comment (arulagrawal) [inline libs/api/request-access/service/feature/src/lib/request-access.service.spec.ts:26]: ?

**Review by arulagrawal — COMMENTED** [review]
I think I'm being to pedantic about style so I'll leave that for now, but the GetAccessEntitiesQuery import is definitely unused. We actually have a lot of things we aren't using, like the query bus variable. But I suppose that's for future use?

**Comment by FaraiQC** [comment]
> I think I'm being to pedantic about style so I'll leave that for now, but the GetAccessEntitiesQuery import is definitely unused. We actually have a lot of things we aren't using, like the query bus variable. But I suppose that's for future use?

Yeah it is for future use since we are still using mock data

**Review by arulagrawal — APPROVED** [review]
